dm/dm/worker/server.go

```diff
@@ -441,30 +441,33 @@ func (s *Server) observeSourceBound(ctx context.Context, rev int64) error {
 }
 
 func (s *Server) doClose() {
-	s.cancel()
-	// close server in advance, stop receiving source bound and relay bound
-	s.wg.Wait()
```

Comment:
this WaitGroup makes sure "stop receiving source bound" happens before "getSourceWorker and Close". Your modification breaks it.